src/vcpkg/base/downloads.cpp

```diff
@@ -883,6 +884,110 @@ namespace vcpkg
         return true;
     }
 
+    bool store_to_azblob_cache(DiagnosticContext& context,
+                               const Filesystem& fs,
+                               StringView raw_url,
+                               const SanitizedUrl& sanitized_url,
+                               View<std::string> headers,
+                               const Path& file,
+                               size_t file_size,
+                               size_t max_single_write,
+                               size_t block_size)
+    {
+        if (file_size <= max_single_write)
+            return store_to_asset_cache(context, raw_url, sanitized_url, "PUT", headers, file);
+
+        static constexpr StringLiteral guid_marker = "63ba93be-2084-4751-8b4b-d481c80d965c";
+
+        // base64 encoded IDs, generated with
+        // for I in @ A B C D E F G H I; do echo '"'$(echo "00${I}" | base64)'",' ; done
+        static const char* block_ids[10] = {
+            "MDBACg==",
+            "MDBBCg==",
+            "MDBCCg==",
+            "MDBDCg==",
+            "MDBECg==",
+            "MDBFCg==",
+            "MDBGCg==",
+            "MDBHCg==",
+            "MDBICg==",
+            "MDBJCg==",
+        };
+
+        if (file_size > block_size * std::extent<decltype(block_ids)>::value) return false;
+
+        auto http_cmd = Command{"curl"}.string_arg("--fail-early");
+
+        auto block_url = raw_url.to_string();
+        block_url += raw_url.contains('?') ? '&' : '?';
+        block_url += "&comp=block&blockid=";
+
+        auto block_file = fmt::format("{}.chunk", file);
+        auto block_id = std::begin(block_ids);
+        auto block_list = std::string("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<BlockList>\n");
+        for (size_t range_from = 0; range_from < file_size; range_from += block_size, ++block_id)
+        {
+            auto range_to = std::min(range_from + block_size, file_size);
+
+            // Extract block data locally
+            http_cmd.string_arg("-o").string_arg(block_file);
+            http_cmd.string_arg("--range").string_arg(fmt::format("{}-{}", range_from, range_to - 1));
+            http_cmd.string_arg(fmt::format("file://{}", file));
```

Comment:
The `file://` syntax needs attention on windows where paths may have `\` separators.

Reply:
this work on windows.